### PR TITLE
Adds general project documentation.

### PR DESCRIPTION
**Pull request by tsroten**
## Description

This updates the README and documentation to provide more information about the project.

## Checklist

- [ ] ~I've added this contribution to the `CHANGELOG`.~ not applicable
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

**Comment by tsroten** [comment]
Here is the documentation link for this branch: http://cli-helpers.readthedocs.io/en/tsroten-general_docs/

**Comment by tsroten** [comment]
@amjith What do you think of the project description/readme/doc wording that I've put in this pull request?

File: README.rst [hunk 1]
```diff
@@ -8,15 +8,25 @@ CLI Helpers
...
+CLI Helpers is not focused on your app's design pattern or framework -- you can
+use it on its own or in combination with other libraries. It's lightweight and
+easy to extend.
 
-A special thanks to `Amjith Ramanujam <https://github.com/amjith>`_, the creator
-of pgcli and mycli.
+What's included in CLI Helpers?
+
+- Prettyprinting of tabular data with custom pre-processing
+- *[in progress]* config file reading/writing
```
Comment (amjith) [inline README.rst]: Missing comma. 

```
Prettyprinting of tabular data *[in progress]*, config file management.
```

Comment (tsroten) [inline README.rst]: I think it renders as a list

Comment (tsroten) [inline README.rst]: http://cli-helpers.readthedocs.io/en/tsroten-general_docs/

File: Makefile [hunk 2]
```diff
@@ -1,10 +1,11 @@
 PROJECT = cli_helpers
 
-.PHONY: clean lint test test-all coverage docs dist release
+.PHONY: clean lint lint-fix test test-all coverage docs dist release
 
 help:
 	@echo "clean - remove all build artifacts"
-	@echo "lint - check code changes against PEP8"
+	@echo "lint - check code changes against PEP 8"
+	@echo "lint-fix - automatically fix PEP 8 violations"
```
Comment (amjith) [inline Makefile]: Nice addition.

File: README.rst (hunk 1 above)
Comment (amjith) [inline README.rst]: That makes sense.

**Comment by amjith** [comment]
This looks awesome! Great work. :tulip:

**Pull request merged**